ldap_auth_provider.py

```diff
@@ -347,6 +374,21 @@ class _LdapConfig(object):
             "mail",
         ])
 
+        ldap_config.ad_forest = config.get("ad_forest", False)
+        if ldap_config.ad_forest:
+            if config["attributes"]["uid"].lower() != "userprincipalname":
+                raise Exception(
+                    "Only userPrincipalName is supported "
+                    "as uid property in ad_forest mode"
+                )
+
+            ldap_config.default_domain = config.get("default_domain", None)
+            ldap_config.mxid_domain_separator = config.get("mxid_domain_separator", "/")
+            if ldap_config.mxid_domain_separator not in ("=", "/"):
+                raise Exception(
+                    "Only = or / symbols supported as domain separator"
+                )
```

Comment:
why is this restriction needed?

Reply:
Those 2 symbols are only symbols allowed in mxid, but disallowed in ```userPrincipalName```.  
However let's drop this configuration in favor of binding ```/``` in mxid and using classic ```domain\login``` form as you suggested.